### PR TITLE
XHR interceptor JSON response handling

### LINKED ISSUE #22466: xhrInterceptor response handling fix
The xhrInterceptor handling of responses (xhr-utils#fromStructuredCloneable) needs to be fixed to handle JSON. Currently this is not an issue as viewerRenderTemplate capability is what's currently being used to proxy requests as well as template info but clients may choose not to handle the templating responsibility that comes with viewerRenderTemplate, using only the xhrInterceptor capability.

### PR DESCRIPTION
**Pull request by alabiaga**
Add support for xhrInterceptor JSON response handling.
Fixes #22466

### CODE REVIEW

**Comment by alabiaga** [comment]
cc\ @zhangsu @GoTcWang 

For an example on usage pls see https://github.com/ampproject/amphtml/pull/22538/files#diff-30452aa7d70fce3d8893af2eac00b52eR513

Demonstrating how the viewer would construct the json response to send back to AMP after intercepting the request.

File: src/utils/xhr-utils.js [hunk 1]
```diff
@@ -136,11 +137,18 @@ export function fromStructuredCloneable(response, responseType) {
 
   const isDocumentType = responseType == 'document';
   if (!isDocumentType) {
+    let body = response['body'];
+    // Handle JSON response.
+    if (body && body['json']) {
```
Comment (zhangsu) [inline src/utils/xhr-utils.js]: Just curious: who's nesting a `json` field under `body` today? AFAICT Gmail isn't using this. We just serialize the body (which can only be `application/json` today) into a string and put it directly into the `body` field.

Comment (alabiaga) [reply src/utils/xhr-utils.js]: This is just following the way json is accessed via API .https://developer.mozilla.org/en-US/docs/Web/API/Body/json. We can go your route as well as only application/json is supported but I wanted to do this incase of future support for other data formats.

Comment (alabiaga) [reply src/utils/xhr-utils.js]: Removed nested json field as header content-type can just be used to read format types in the future. Thanks Su.

**Pull request closed without merging**